### ISSUE TITLE
NoClassDefFoundError with Gradle 4.x

### ISSUE THREAD

**Issue by reisub** [johnrengelman/shadow#341]
@johnrengelman first let me just say - thanks for the really great project, it helps a lot and I use it for lots of different projects. You've done a great job!

While I realize the README states compatibility with Gradle 3.x and 4.0 nightlies, I want to add this issue for anyone out there who runs into the same issue while routinely bumping gradle versions or not reading the README carefully enough :)

And maybe someone decides to help out and fix this issue.

This is the exception I get:

```
java.lang.NoClassDefFoundError: Could not initialize class org.eclipse.jetty.server.HttpConnection
    at org.eclipse.jetty.server.HttpConnectionFactory.newConnection(HttpConnectionFactory.java:85)
    at org.eclipse.jetty.server.ServerConnector$ServerConnectorManager.newConnection(ServerConnector.java:482)
    at org.eclipse.jetty.io.ManagedSelector.createEndPoint(ManagedSelector.java:424)
    at org.eclipse.jetty.io.ManagedSelector.access$1600(ManagedSelector.java:61)
    at org.eclipse.jetty.io.ManagedSelector$CreateEndPoint.run(ManagedSelector.java:603)
    at org.eclipse.jetty.util.thread.Invocable.invokePreferred(Invocable.java:128)
    at org.eclipse.jetty.util.thread.Invocable$InvocableExecutor.invoke(Invocable.java:222)
    at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:294)
    at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:199)
    at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:673)
    at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:591)
    at java.lang.Thread.run(Thread.java:745)
```

Please check the (User Guide)[http://imperceptiblethoughts.com/shadow] before submitting "how do I do 'x'?" questions!

### Shadow Version

2.0.1

### Gradle Version

4.x

### Expected Behavior

Built shadow jar works like on earlier Gradle versions since it builds without error.

### Actual Behavior

`NoClassDefFoundError` is thrown at runtime, although the referenced class is contained within the shadow jar.

### Gradle Build Script(s)

https://gist.github.com/reisub/c7b1570f10a60b33312d7d22c7f447d4

### Content of Shadow JAR (`jar tf <jar file>` - post link to GIST if too long)

https://gist.github.com/reisub/97684e38b69113d9bce7f31f90c802bc

**Comment by reisub** [comment]
I found the solution here:
https://stackoverflow.com/a/48698984/2643666

TL;DR adding this into my `build.gradle` fixed the problem:

```
shadowJar {
    transform(ServiceFileTransformer) {
        path = 'META-INF/services'
        include 'org.eclipse.jetty.http.HttpFieldPreEncoder'
    }
}
```

**Comment by uuf6429** [comment]
I have this same problem. The fix above causes a compile error:
```
> Could not get unknown property 'ServiceFileTransformer' for task ':shadowJar' of type com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar.
```
